dynamic/discovery/discovery.go

```diff
@@ -139,34 +135,24 @@ func (rm *ResourceMap) refresh() {
 			// Remember how to map back from Kind/subresource to resource.
 			// This is different from what RESTMapper provides because we already know
 			// the full GroupVersionKind and just need the resource name.
+			// Remember which resources are subresources, and map the kind to the main resource.
 			if strings.ContainsRune(apiResource.Name, '/') {
-				// Remember which resources are subresources, and map the kind to the main resource.
 				gve.subresources[apiResource.Name] = apiResource
 			} else {
-				// Make sure we don't choose a subresource like "pods/status".
 				gve.kinds[apiResource.Kind] = apiResource
 			}
 		}
 
 		// Group all subresources for a resource.
-		for i := range group.APIResources {
-			apiResource := &APIResource{
-				APIResource: group.APIResources[i],
-				APIVersion:  group.GroupVersion,
-			}
-			if !strings.ContainsRune(apiResource.Name, '/') {
-				apiResource.SubresourceMap = make(map[string]bool)
-				for _, apiSubresource := range gve.subresources {
-					split := strings.Split(apiSubresource.Name, "/")
-					resourceName := split[0]
-					subresourceKey := split[1]
-					if apiResource.Name == resourceName {
-						apiResource.SubresourceMap[subresourceKey] = true
-					}
-				}
-				gve.resources[apiResource.Name] = apiResource
-				gve.kinds[apiResource.Kind] = apiResource
+		for apiSubresourceName, _ := range gve.subresources {
```

Comment:
@enisoc All tests passed now. Couldn't we achieve the same by adding the logic inside `if strings.ContainsRune(apiResource.Name, '/') {...}`?